### PR TITLE
Modify pattern for device bit check on s390x

### PR DESCRIPTION
**Pull request by fbq815**
Virtual_block: Add id_pattern works on s390x and modify the checking
command to adapt the test on s390x

ID:1969638

Signed-off-by: Boqiao Fu <bfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
@PaulYuuu @ngu-niny @zhenyzha  please help me review this patch

**Comment by fbq815** [comment]
Test result:
JOB ID     : 3ed8a6fa562170416205d4c448e1941680c0b174
JOB LOG    : /home/bfu/kar/workspace/job-results/job-2021-06-08T15.40-3ed8a6f/job.log
 (1/1) Host_RHEL.m8.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.s390x.io-github-autotest-qemu.device_bit_check.block_device.s390-virtio: PASS (87.09 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/bfu/kar/workspace/job-results/job-2021-06-08T15.40-3ed8a6f/results.html
JOB TIME   : 87.45 s

File: qemu/tests/device_bit_check.py [hunk 1]
```diff
@@ -72,11 +73,24 @@ def run(test, params, env):
...
             logging.info("Properity bit in qtree is right for %s.", option)
             if params.get("check_in_guest", "yes") == "yes":
-                pci_info = session.cmd_output("lspci")
-                pci_n = re.findall(pci_id_pattern, pci_info)
-                if not pci_n:
-                    test.error("Can't get the pci id for device")
-                cmd = "cat /sys/bus/pci/devices/0000:%s/" % pci_n[0]
+                if params.get('machine_type').startswith("s390"):
+                    ccw_id_pattern = \
+                        ccw_id_pattern + '%s' % \
```
Comment (PaulYuuu) [inline qemu/tests/device_bit_check.py:78]: Why not `ccw_id_pattern + re.findallxxxx`

File: qemu/tests/device_bit_check.py [hunk 2]
```diff
@@ -72,11 +73,24 @@ def run(test, params, env):
 
             logging.info("Properity bit in qtree is right for %s.", option)
             if params.get("check_in_guest", "yes") == "yes":
-                pci_info = session.cmd_output("lspci")
-                pci_n = re.findall(pci_id_pattern, pci_info)
-                if not pci_n:
-                    test.error("Can't get the pci id for device")
-                cmd = "cat /sys/bus/pci/devices/0000:%s/" % pci_n[0]
+                if params.get('machine_type').startswith("s390"):
+                    ccw_id_pattern = \
```
Comment (PaulYuuu) [inline qemu/tests/device_bit_check.py:77]: You should use another namespace here, then L87 will become unnecessary.

File: qemu/tests/device_bit_check.py (hunk 1 above)
Comment (fbq815) [reply qemu/tests/device_bit_check.py:78]: well, I think this would be better to read, modified

File: qemu/tests/device_bit_check.py [hunk 3]
```diff
@@ -72,11 +73,24 @@ def run(test, params, env):
...
+                        ccw_id_pattern + '%s' % \
+                        re.findall('dev:virtio-scsi-ccw.*\n'
+                                   '.*\n.*\n.*\ndev_id=\"fe.0.(.*?)\"',
+                                   qtree_info.replace(' ', ''))[0]
+                    ccw_info = session.cmd_output("lscss")
+                    ccw_n = re.findall(ccw_id_pattern, ccw_info)
+                    if not ccw_n:
+                        test.error("Can't get the ccw id for device")
+                    cmd = "cat /sys/bus/ccw/devices/%s/" % ccw_n[0]
+                    ccw_id_pattern = params.get("ccw_id_pattern")
```
Comment (PaulYuuu) [inline qemu/tests/device_bit_check.py:87]:
```suggestion
                    id_pattern = (ccw_id_pattern + 
                                  re.findall('dev:virtio-scsi-ccw.*\n*\n.*\n'
                                             '.*\ndev_id=\"fe.0.(.*?)\"',
                                             qtree_info.replace(' ', ''))[0])
                    ccw_info = session.cmd_output("lscss")
                    ccw_n = re.findall(id_pattern, ccw_info)
                    if not ccw_n:
                        test.error("Can't get the ccw id for device")
                    cmd = "cat /sys/bus/ccw/devices/%s/" % ccw_n[0]
```

Comment (fbq815) [reply qemu/tests/device_bit_check.py]: Modified

File: qemu/tests/device_bit_check.py (hunk 2 above)
Comment (fbq815) [reply qemu/tests/device_bit_check.py]: Modified

**Comment by fbq815** [comment]
Test result:
JOB ID     : 8476173a4abeb85326eca0b0f5f4c14de24d6e7d
JOB LOG    : /home/bfu/kar/workspace/job-results/job-2021-06-16T03.34-8476173/job.log
 (1/1) Host_RHEL.m8.u4.product_av.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.s390x.io-github-autotest-qemu.device_bit_check.block_device.s390-virtio: PASS (80.78 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/bfu/kar/workspace/job-results/job-2021-06-16T03.34-8476173/results.html
JOB TIME   : 81.11 s

**Comment by ngu-niny** [comment]
LGTM.

**Comment by zhenyzha** [comment]
git am 2851.patch 
Applying: Modify pattern for device bit check on s390x

No regression on aarch64:
```
 (1/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.aarch64.io-github-autotest-qemu.device_bit_check.block_device.arm64-pci: PASS (203.68 s)
 (2/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.aarch64.io-github-autotest-qemu.device_bit_check.nic_device.arm64-pci: PASS (218.83 s)
```
No regression on ppc64le:
```
 (1/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.device_bit_check.block_device: PASS (335.24 s)
 (2/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.device_bit_check.nic_device: PASS (337.50 s)
```
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**